torch/csrc/jit/script/compiler.cpp

```diff
@@ -836,32 +836,194 @@ struct to_ir {
       , graph(method.graph())
       , def(def)
       , resolver(std::move(resolver_))
+      , self(self)
       , environment_stack(nullptr) {
     JIT_ASSERT(resolver);
     pushFrame(graph->block());
 
-    auto schema = extractSchemaFromDef(def, bool(self));
-
-    std::vector<Argument> arguments, returns; // for schema
-    // inputs
-    auto it = def.decl().params().begin();
-    auto end = def.decl().params().end();
     // Type annotations exclude explicitly typing the "self" parameter, so in the
     // case that this is a method with self we expect one fewer parameter annotation
     // than the number of parameters this Def takes.
     if (self && def.decl().params().size() == 0) {
       throw ErrorReport(def.decl().params().range()) << "methods must have a self argument";
     }
+
```

Comment:
note: most of this is just refactoring the body of the constructor into emitFormalArguments and emitReturn. The new functionality is in evaluateDefaults.